### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by kripper** [EpistasisLab/KRAGEN#96]
For the [benchmark](https://github.com/EpistasisLab/KRAGEN/blob/main/Benchmark_Comparison_and_Setup.md), are you only evaluating the GoT and React techniques?

Or is are knowledge graph and RAG techniques also being tested? If so, how are you using the knowledge graph in this case?

**Comment by HyunjunA** [comment]
Thank you for your question. Regarding the benchmarking, we compared the performance of KRAGEN (GoT + RAG) combined with ChatGPT, ChatGPT without KRAGEN, and two open-source LLM models without KRAGEN on datasets specifically created for our research. KRAGEN employs a combination of Graph of Thought (GoT) and Retrieval-Augmented Generation (RAG) techniques to answer user queries.

For Alzheimer's information, we converted the data in The Alzheimer's KnowledgeBase (AlzKB) into vectors and stored them in a vector database, which is then used for Retrieval-Augmented Generation (RAG). Additionally, the REACT technique was used for developing the web app in this project.

I hope this clarifies your query. If you have any further questions, please feel free to ask.

Thank you.

**Comment by kripper** [comment]
Thanks for answering.
Yes, I have a specific question.

> we converted the data in The Alzheimer's KnowledgeBase (AlzKB) into vectors and stored them in a vector database

Did you use a knowledge graph here? If so, how did you build it?

**Comment by HyunjunA** [comment]
Yes, a knowledge graph was used in the study. This knowledge graph was created by my colleagues. The Alzheimer's Knowledge Base (AlzKB) is designed as a large, heterogeneous graph knowledge base. It was assembled using 22 diverse external data sources that describe biological and pharmaceutical entities at different levels of organization, such as chemicals, genes, anatomy, and diseases.

For more detailed information, please refer to the paper below:
https://www.jmir.org/2024/1/e46777/

**Comment by kripper** [comment]
Thanks. The knowledge graph was created manually. Have you considered any tool to automate the knowledge graph generation process for using together with Kragen?

**Comment by HyunjunA** [comment]
No problem. It is an interesting idea, I think one of my colleagues has been working on it!
